### ISSUE TITLE
[Feature] Analyzer can analyze enum members

### ISSUE THREAD

**Issue by xXTurnerLP** [dnSpyEx/dnSpy#145]
It will be cool if the analyzer can show where an enum member is used.

For example, currently this is the view you get when you analyze an enum member:
![image](https://user-images.githubusercontent.com/40324267/204142498-87aa4ab6-e113-434b-a20d-76f04addbacf.png)

And this is the enum
![image](https://user-images.githubusercontent.com/40324267/204142518-a7bf7e39-527a-4f11-bc84-4c671f94163f.png)

It will be cool if you could see where that F_OK was either set or read

EDIT: The above sentence may sound confusing, I meant if you had a variable that holds such an enum you could search that the enum type was used in such a variable, see example below:

```cs
public AccessMode mode;

public void SomeFunction()
{
    mode = AccessMode.R_OK // this should get detected by the analyzer
}

public bool SomeFunction2()
{
    if (mode == AccessMode.X_OK) // this should get detected by the analyzer
        return true;
    else
        return false;
}
```

**Comment by sunnamed434** [comment]
Great feature

**Comment by ElektroKill** [comment]
Duplicate of https://github.com/dnSpyEx/dnSpy/issues/50

TLDR: Implementing this is rather hard as enum fields are not loaded via a `ldsfld` instruction which is usually used to load fields but rather a `ldc.i4` instruction which pushes a raw number to the stack. The decompiler is able to recover this information by performing several passes of type inference/analysis for the stack values based on local types, parameter types, return types, etc. Type analysis is a complex and computationally intensive process and implementing it in the analyzer is hard. Especially since the analyzer only has access to the low-level IL code while the decompiler uses a higher-level ILAst instruction model when it performs type inference. Building the ILAst is also a complex and computationally intensive process.

Right now the only way you can search for enum constants is by switching the search window mode to `Number/String` and searching for the literal constant of the enum. For example, `1` for `R_OK`.

Due to the complexity of such an analyzer, it will not be added any time soon. Sorry!

**Comment by xXTurnerLP** [comment]
Ok, thanks for the tip!
